pkg/daemon/daemon.go

```diff
@@ -214,23 +214,46 @@ func (dn *Daemon) addProfileConfig(socketPath string, nodeProfile *ptpv1.PtpProf
 		nodeProfile.Ptp4lOpts = &opts
 	}
 
+	output := new(ptp4lconf)
 	if nodeProfile.Interface == nil || *nodeProfile.Interface == "" {
-		// We need to copy this to another variable because is a pointer
 		iface := string("")
 		nodeProfile.Interface = &iface
+	} else {
+		output.ifaceString = fmt.Sprintf("[%s]", *nodeProfile.Interface)
 	}
 
-	config := fmt.Sprintf("%s\nuds_address %s\nmessage_tag [%s]",
-		*nodeProfile.Ptp4lConf,
-		socketPath,
-		*nodeProfile.Interface)
-	nodeProfile.Ptp4lConf = &config
+	output.populatePtp4lConf(nodeProfile.Ptp4lConf)
+
+	if nodeProfile.Interface != nil && *nodeProfile.Interface != "" {
+		ifaceSection := fmt.Sprintf("[%s]\n", *nodeProfile.Interface)
+		output.sections[ifaceSection] = *new(ptp4lconfSection)
+	}
+
+	for n, line := range output.sections["[global]"].options{
+		if strings.HasPrefix(line, "uds_address") {
+			output.sections["[global]"].options[n] = ""
+		}
+
+	}
+
+
+	uds_address := fmt.Sprintf("uds_address %s\n", socketPath)
+	message_tag := fmt.Sprintf("message_tag [%s]\n", *nodeProfile.Name)
+	section := output.sections["[global]"]
+	section.options = append(section.options, message_tag)
```

Comment:
use a map as the options this will allow you to find the key (or create of not exist)